tests/maven.Tests.ps1

```diff
@@ -0,0 +1,73 @@
+$SUT_IMAGE="pester-maven"
+$SUT_CONTAINER="pester-maven"
+$SUT_TAG="jdk-8"
+if(![System.String]::IsNullOrWhiteSpace($env:TAG)) {
+  $SUT_TAG=$env:TAG
+}
+$SUT_TEST_IMAGE="pester-maven-test"
+$SUT_TEST_CONTAINER="pester-maven-test"
+
+Import-Module -Force -DisableNameChecking $PSScriptRoot/test_helpers.psm1
+
+Describe "$SUT_TAG build image" {
+  BeforeEach {
+    Push-Location -StackName 'maven' -Path "$PSScriptRoot/../$SUT_TAG"
+  }
+
+  It 'builds image' {
+    $exitCode, $stdout, $stderr = Build-Docker --pull -t $SUT_IMAGE
+    $lastExitCode | Should -Be 0
+  }
+
+  AfterEach {
+    Pop-Location -StackName 'maven'
+  }
+}
+
+Describe "$SUT_TAG build test image" {
+  It 'builds image' {
+    Push-Location -StackName 'maven' -Path $PSScriptRoot
+    $exitCode, $stdout, $stderr = Build-Docker -t $SUT_TEST_IMAGE
+    $exitCode | Should -Be 0
+  }
+}
+
+Describe "$SUT_TAG create test container" {
+  It 'creates test container' {
+    $version = $(Get-Content -Path "$PSScriptRoot/../$SUT_TAG/Dockerfile.windows" | Select-String -Pattern 'ARG MAVEN_VERSION.*' | ForEach-Object { $_ -replace 'ARG MAVEN_VERSION=','' })
+    $exitCode, $stdout, $stderr = Run-Program -Cmd "docker.exe" -Params "run --rm $SUT_IMAGE mvn -version"
+    $exitCode | Should -Be 0
+    $stdout | Should -Match "Apache Maven $version"
+  }
+}
+
+Describe "$SUT_TAG settings.xml is setup" {
+  It 'sets up settings.xml' {
+    $exitCode, $stdout, $stderr = Run-Program -Cmd "docker.exe" -Params "run --rm $SUT_TEST_IMAGE Get-Content C:/Users/ContainerAdministrator/.m2/settings.xml"
+    $exitCode | Should -Be 0
+    "$PSScriptRoot/settings.xml" | Should -FileContentMatchMultiline $stdout
+  }
+}
+
+
+Describe "$SUT_TAG repository is created" {
+  It 'created repository' {
+    $exitCode, $stdout, $stderr = Run-Program -Cmd "docker.exe" -Params "run --rm $SUT_TEST_IMAGE if(Test-Path C:/Users/ContainerAdministrator/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar) { exit 0 } else {exit 1 }"
+    $exitCode | Should -Be 0
+  }
+}
+
+
+Describe "$SUT_TAG run Maven" {
+  It 'runs maven' {
+    $exitCode, $stdout, $stderr = Run-Program -Cmd "docker.exe" -Params "run --rm $SUT_TEST_IMAGE mvn -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -f $env:TEMP install"
```

Comment:
```suggestion
    $exitCode, $stdout, $stderr = Run-Program -Cmd "docker.exe" -Params "run --rm $SUT_TEST_IMAGE mvn -B -ntp -f $env:TEMP install"
```

since we are on 3.6.1